gcp/table_gcp_redis_instance.go

```diff
@@ -0,0 +1,360 @@
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "persistence_config",
+				Description: "Persistence configuration parameters.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "suspension_reasons",
+				Description: "Reasons that causes instance in `SUSPENDED` state.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "maintenance_version",
+				Description: "The self service update maintenance version.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "available_maintenance_versions",
+				Description: "The available maintenance versions that an instance could update to.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(gcpRedisInstanceTurbotData, "Akas"),
+			},
+
+			// standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     plugin.HydrateFunc(getProject).WithCache(),
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTIONS
+
+func listGcpRedisInstances(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	// Create Service Connection
+	service, err := RedisService(ctx, d)
+	if err != nil {
+		logger.Error("gcp_redis_instance.listGcpRedisInstances", "connection_error", err)
+		return nil, err
+	}
+
+	var location string
+	matrixLocation := d.EqualsQualString(matrixKeyRedisLocation)
+	// Since, when the service API is disabled, matrixLocation value will be nil
+	if matrixLocation != "" {
+		location = matrixLocation
+	}
+
+	getProjectCached := plugin.HydrateFunc(getProject).WithCache()
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		logger.Error("gcp_redis_instance.listGcpRedisInstances", "cache_error", err)
+		return nil, err
+	}
+	project := projectId.(string)
+
+	parent := "projects/" + project + "/locations/" + location
+	req := &redispb.ListInstancesRequest{
+		Parent: parent,
+	}
+
+	it := service.ListInstances(ctx, req)
+	for {
+		resp, err := it.Next()
+		if err != nil {
+			if err == iterator.Done {
+				break
+			}
+			logger.Error("gcp_redis_instance.listGcpRedisInstances", "api_error", err)
+			return nil, err
+		}
+		test := resp.CreateTime.AsTime()
+		fmt.Println(test)
+		d.StreamListItem(ctx, resp)
+
+		if d.RowsRemaining(ctx) == 0 {
```

Comment:
```suggestion
                	// Check if context has been cancelled or if the limit has been hit (if specified)
		// if there is a limit, it will return the number of rows required to reach this limit
		if d.RowsRemaining(ctx) == 0 {
```